_delphi_utils_python/delphi_utils/validator/report.py

```diff
@@ -73,12 +73,15 @@ def add_raised_warning(self, warning):
         """
         self.raised_warnings.append(warning)
 
-    def summary(self):
+    def summary(self, logger=None):
         """Represent summary of report as a string."""
         out_str = f"{self.total_checks} checks run\n"
         out_str += f"{len(self.unsuppressed_errors)} checks failed\n"
         out_str += f"{self.num_suppressed} checks suppressed\n"
         out_str += f"{len(self.raised_warnings)} warnings\n"
+        if logger is not None:
+            logger.info(out_str)
+        print(out_str)
         return out_str
```

Comment:
This needs a big of rearranging. I'd just save `out_str` to an attribute of ValidationReport here, and add it to the log in the `log()` method. (The `summary` attribute and `set_summary` method names are just the first things that came to mind, but may not be the best options.)

```suggestion
    def set_summary(self):
        """Represent summary of report as a string."""
        out_str = f"{self.total_checks} checks run\n"
        out_str += f"{len(self.unsuppressed_errors)} checks failed\n"
        out_str += f"{self.num_suppressed} checks suppressed\n"
        out_str += f"{len(self.raised_warnings)} warnings\n"
        self.summary = out_str
```

`log()` already handles the case where the logger is `None`, so we don't want to have to handle that logic again. In fact, if the logger is `None`, `log()` gets a new logger, but the way this is set up now, `summary()` won't be able to access it because `summary()` is run before `log()` in `print_and_exit()`.

Reply:
I believe that I'll need to add logger.info() to the log() function then?

Reply:
>  I'll need to add logger.info() to the log() function then?

yep